src/lib/storage/base_attribute_vector.hpp

```diff
@@ -26,5 +28,7 @@ class BaseAttributeVector : private Noncopyable {
 
   // returns the width of the values in bytes
   virtual AttributeVectorWidth width() const = 0;
+
+  virtual std::shared_ptr<BaseAttributeVector> migrate(const PolymorphicAllocator<size_t>& alloc) const = 0;
```

Comment:
Yes, it is technically a copy. My migration model is: Keep the chunk object, but copy all subordinate objects (columns, attribute vectors) to a new NUMA node.

Reply:
Can we come up with a better name then? "migrate" sounds as if the data is gone in the original place.

Reply:
How about `copy_with_allocator` or just `copy`?

Reply:
I like `copy_using_allocator` a tad better, because it makes it clear that the allocator is not being copied. Also, a short comment would be great that this can be used to migrate things to a different NUMA node if that's what the other allocator does.